src/features/device-state/routes/state-get-v3.ts

```diff
@@ -108,24 +108,32 @@ function buildAppFromRelease(
 		// extract the per-image information
 		const image = ipr.image[0];
 
-		const si = serviceInstallFromImage(device, image);
+		const si = serviceInstallFromImage(device ?? application, image);
```

Comment:
Oh, sorry, I see that you are checking the `device` below to add device environment variables and other things. 

I don't really like this pattern, but I since so few things change in the function body depending on whether is an application or a device that I think it's fine.

Reply:
The idea is an as close as possible overlap with the device target state v3 code. Hence, we decided to go this way instead of introducing redundant code.